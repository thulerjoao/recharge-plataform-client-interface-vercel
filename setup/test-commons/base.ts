import type {Config} from '@jest/types'
import path from 'path'

//Sync object
const config: Config.InitialOptions = {
    testEnvironment: 'jsdom',
    verbose: true,
    setupFilesAfterEnv: [
        path.resolve(__dirname, 'src', 'setupTests.js')
    ],
    transform: {
        '\\.[jt]sx?$': ['babel-jest', {
            configFile: path.resolve(__dirname, '.babelrc')
        }],
    },
    moduleNameMapper: {
      '^axios$': require.resolve('axios'),
    },
    testMatch: [
      '**/__tests__/**/*.[jt]s?(x)',
      '**/?(*.)+(spec|test).[tj]s?(x)',
    ],
}
export default config

