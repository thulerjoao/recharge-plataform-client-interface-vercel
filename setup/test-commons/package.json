{
    "name": "@4miga/test-commons",
    "version": "1.0.0",
    "private": true,
    "main": "index.js",
    "scripts": {
      "lint": "echo 'Not necessary'",
      "test": "echo 'Not necessary'"
    },
    "files":[
      "base.ts"
    ],
    "devDependencies": {
        "jest":"27.0.0",
        "ts-node":"10.7.0",
        "babel-jest":"28.0.3",
        "@babel/core":"7.17.10",
        "@babel/preset-env":"7.17.10",
        "@babel/preset-react":"7.16.7",
        "@babel/preset-typescript":"7.16.7",
        "@testing-library/react": "14.2.1",
        "jest-styled-components" : "7.2.0"
    }
  }
