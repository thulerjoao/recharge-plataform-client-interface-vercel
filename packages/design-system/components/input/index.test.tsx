import { customRender } from "@4miga/test-commons/react-testing-library";
import Input from "./index";

const render = customRender();

describe("Global input snapshot test", () => {
  const heights = [40, 48] as const;

  heights.forEach((height) => {
    test(`matches snapshot with height ${height}`, () => {
      const { asFragment } = render(
        <Input height={height} value={`The height value is ${height}`} />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  test("matches the snapshot with a left element", () => {
    const { asFragment } = render(
      <Input height={40} leftElement={<span>Left</span>} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches the snapshot when loading", () => {
    const { asFragment } = render(<Input height={40} loading={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches the snapshot with custom margin and padding", () => {
    const { asFragment } = render(
      <Input height={48} margin="10px" padding="5px" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
