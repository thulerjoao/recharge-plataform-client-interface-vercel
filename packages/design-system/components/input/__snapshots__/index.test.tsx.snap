// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global input snapshot test matches snapshot with height 40 1`] = `
<DocumentFragment>
  .c1 {
  box-sizing: border-box;
  padding: 0px 16px;
  width: 100%;
  font-size: 16px;
  height: 36px;
  border-radius: 8px;
  border: 2px solid #525F63;
  color: #525F63;
}

.c1::placeholder {
  color: #525F63;
}

.c1:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: auto;
}

<div
    class="c0"
  >
    <input
      class="c1"
      height="40"
    />
  </div>
</DocumentFragment>
`;

exports[`Global input snapshot test matches snapshot with height 48 1`] = `
<DocumentFragment>
  .c1 {
  box-sizing: border-box;
  padding: 0px 16px;
  width: 100%;
  font-size: 16px;
  height: 44px;
  border-radius: 8px;
  border: 2px solid #525F63;
  color: #525F63;
}

.c1::placeholder {
  color: #525F63;
}

.c1:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: auto;
}

<div
    class="c0"
  >
    <input
      class="c1"
      height="48"
    />
  </div>
</DocumentFragment>
`;

exports[`Global input snapshot test matches the snapshot when loading 1`] = `
<DocumentFragment>
  .c1 {
  box-sizing: border-box;
  padding: 0px 16px;
  width: 100%;
  font-size: 16px;
  height: 36px;
  border-radius: 8px;
  border: 2px solid #525F63;
  color: #525F63;
}

.c1::placeholder {
  color: #525F63;
}

.c1:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: auto;
}

<div
    class="c0"
  >
    <input
      class="c1"
      height="40"
    />
  </div>
</DocumentFragment>
`;

exports[`Global input snapshot test matches the snapshot with a left element 1`] = `
<DocumentFragment>
  .c2 {
  box-sizing: border-box;
  padding: 0px 16px;
  width: 100%;
  font-size: 16px;
  height: 36px;
  border-radius: 8px;
  border: 2px solid #525F63;
  color: #525F63;
}

.c2::placeholder {
  color: #525F63;
}

.c2:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: auto;
}

.c1 {
  position: absolute;
  left: 8px;
  top: 0;
  width: 16px;
  height: 100%;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span>
        Left
      </span>
    </div>
    <input
      class="c2"
      height="40"
    />
  </div>
</DocumentFragment>
`;

exports[`Global input snapshot test matches the snapshot with custom margin and padding 1`] = `
<DocumentFragment>
  .c1 {
  box-sizing: border-box;
  padding: 0px 16px;
  padding: 5px;
  width: 100%;
  font-size: 16px;
  height: 44px;
  border-radius: 8px;
  border: 2px solid #525F63;
  color: #525F63;
}

.c1::placeholder {
  color: #525F63;
}

.c1:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: auto;
}

<div
    class="c0"
  >
    <input
      class="c1"
      height="48"
      padding="5px"
    />
  </div>
</DocumentFragment>
`;
